### DISCUSSION TITLE
Recommendations when hosting the same app multiple times on the same server

### DISCUSSION THREAD

**Discussion by fbourigault** [symfony/symfony#43119] [Q&A]
We are hosting the same app in different directories on the same server.

Each app instance may be using a different version of the code but each app is configured as a pool in the same php-fpm instance.

We are experiencing weird behavior between app instances.

Mostly validation having an old behavior on an up-to-date instance.

Also, the app cache is warmed-up in our CI Server and we are deploying the build result as an archive.

What are the recommendations in such case, especially for caching (which I suspect to be the root cause of the strange behavior we noticed) ?

EDIT :

After further investigation and a deep look at how `ApcuAdapter` works, I think I should set the `framework.cache.prefix_seed` to the domain name of the app instance through an environment variable. So there will be a stable namespace per instance and each time the `container.build_id` will change, the whole namespace will be cleared so starting with a fresh cache after each deployment.

Am I right?

**Comment by stof** [comment]
Well, first, make sure you don't use preloading, as it might use the preloaded code of a different version. IIRC, php-fpm does not allow to have different preload config per pool.

**Reply by fbourigault** [reply]
We don't have preloading enabled. But "traditional" opcache and apcu are enabled.

Should I use an environment variable in `framework.cache.prefix_seed` to be sure to have something different per app?
